In [56]:
#Importing important libraries
import pymongo
import pandas as pd
import json
import numpy as np
from pymongo import MongoClient

In [57]:
#Importing the dataset and reading it
fligts_statistics = pd.read_csv("Air_Traffic_Landings_Statistics - Copy.csv")
fligts_statistics.head()

Activity Period Operating Airline Operating Airline IATA Code  \
0           200507           ABX Air                          GB   
1           200507           ABX Air                          GB   
2           200507      ATA Airlines                          TZ   
3           200507      ATA Airlines                          TZ   
4           200507        Air Canada                          AC   

  Published Airline Published Airline IATA Code    GEO Summary GEO Region  \
0           ABX Air                          GB       Domestic         US   
1           ABX Air                          GB       Domestic         US   
2      ATA Airlines                          TZ       Domestic         US   
3      ATA Airlines                          TZ       Domestic         US   
4        Air Canada                          AC  International     Canada   

  Landing Aircraft Type Aircraft Body Type Aircraft Manufacturer  \
0             Freighter        Narrow Body     McDonnell Douglas   
1             Freighter        Narrow Body     McDonnell Douglas   
2             Passenger        Narrow Body                Boeing   
3             Passenger        Narrow Body                Boeing   
4             Passenger          Wide Body                Boeing   

  Aircraft Model Aircraft Version  Landing Count  Total Landed Weight  
0           DC-9               30             40              4066000  
1           DC-9               41              1               102000  
2            757              200              2               396000  
3            757              300            167             37408000  
4            767              333              1               320000

In [58]:
#Connecting to mongo db
db_client = pymongo.MongoClient("mongodb://localhost:27017/")
#Creating the database
db = db_client.Flight_details
collection = db.test_collection
document=collection.insert_many(fligts_statistics.to_dict('r'))

C:\Users\gogat\AppData\Local\Temp\ipykernel_16340\3847226230.py:6: FutureWarning: Using short name for 'orient' is deprecated. Only the options: ('dict', list, 'series', 'split', 'records', 'index') will be used in a future version. Use one of the above to silence this warning.
  document=collection.insert_many(fligts_statistics.to_dict('r'))


In [59]:
pip install pymongo

Note: you may need to restart the kernel to use updated packages.


In [60]:
#Just making sure that, its saved in DB
x = collection.find_one()
print(x)

{'_id': ObjectId('644a600c6b7d541964d8758d'), 'Activity Period': 200507, 'Operating Airline': 'ABX Air', 'Operating Airline IATA Code': 'GB', 'Published Airline': 'ABX Air', 'Published Airline IATA Code': 'GB', 'GEO Summary': 'Domestic', 'GEO Region': 'US', 'Landing Aircraft Type': 'Freighter', 'Aircraft Body Type': 'Narrow Body', 'Aircraft Manufacturer': 'McDonnell Douglas', 'Aircraft Model': 'DC-9', 'Aircraft Version': '30', 'Landing Count': 40, 'Total Landed Weight': 4066000}


In [61]:
#Fetching the dataset from mongodb into dataframes
Data_from_db = pd.DataFrame(list(collection.find()))
Data_from_db

_id  Activity Period     Operating Airline  \
0      644a600c6b7d541964d8758d           200507               ABX Air   
1      644a600c6b7d541964d8758e           200507               ABX Air   
2      644a600c6b7d541964d8758f           200507          ATA Airlines   
3      644a600c6b7d541964d87590           200507          ATA Airlines   
4      644a600c6b7d541964d87591           200507            Air Canada   
...                         ...              ...                   ...   
57376  644a600c6b7d541964d955ad           202212  Vietnam Airlines JSC   
57377  644a600c6b7d541964d955ae           202212       Virgin Atlantic   
57378  644a600c6b7d541964d955af           202212      WestJet Airlines   
57379  644a600c6b7d541964d955b0           202212      WestJet Airlines   
57380  644a600c6b7d541964d955b1           202212      WestJet Airlines   

      Operating Airline IATA Code     Published Airline  \
0                              GB               ABX Air   
1                              GB               ABX Air   
2                              TZ          ATA Airlines   
3                              TZ          ATA Airlines   
4                              AC            Air Canada   
...                           ...                   ...   
57376                          VN  Vietnam Airlines JSC   
57377                          VS       Virgin Atlantic   
57378                          WS      WestJet Airlines   
57379                          WS      WestJet Airlines   
57380                          WS      WestJet Airlines   

      Published Airline IATA Code    GEO Summary GEO Region  \
0                              GB       Domestic         US   
1                              GB       Domestic         US   
2                              TZ       Domestic         US   
3                              TZ       Domestic         US   
4                              AC  International     Canada   
...                           ...            ...        ...   
57376                          VN  International       Asia   
57377                          VS  International     Europe   
57378                          WS  International     Canada   
57379                          WS  International     Canada   
57380                          WS  International     Canada   

      Landing Aircraft Type Aircraft Body Type Aircraft Manufacturer  \
0                 Freighter        Narrow Body     McDonnell Douglas   
1                 Freighter        Narrow Body     McDonnell Douglas   
2                 Passenger        Narrow Body                Boeing   
3                 Passenger        Narrow Body                Boeing   
4                 Passenger          Wide Body                Boeing   
...                     ...                ...                   ...   
57376             Passenger          Wide Body                Airbus   
57377             Passenger          Wide Body                Boeing   
57378             Passenger        Narrow Body                Boeing   
57379             Passenger        Narrow Body                Boeing   
57380             Passenger        Narrow Body                Boeing   

      Aircraft Model Aircraft Version  Landing Count  Total Landed Weight  
0               DC-9               30             40              4066000  
1               DC-9               41              1               102000  
2                757              200              2               396000  
3                757              300            167             37408000  
4                767              333              1               320000  
...              ...              ...            ...                  ...  
57376           A359                -             17              7758069  
57377           B789                -             31             13175000  
57378           B38M                -              5               764000  
57379           B737                -           

In [62]:
#We could see there is a _id field which is system generated, as we dont need it, will fetch the values excluding it.
Data_from_db = pd.DataFrame(list(collection.find({}, {'_id': 0})))
Data_from_db

Activity Period     Operating Airline Operating Airline IATA Code  \
0               200507               ABX Air                          GB   
1               200507               ABX Air                          GB   
2               200507          ATA Airlines                          TZ   
3               200507          ATA Airlines                          TZ   
4               200507            Air Canada                          AC   
...                ...                   ...                         ...   
57376           202212  Vietnam Airlines JSC                          VN   
57377           202212       Virgin Atlantic                          VS   
57378           202212      WestJet Airlines                          WS   
57379           202212      WestJet Airlines                          WS   
57380           202212      WestJet Airlines                          WS   

          Published Airline Published Airline IATA Code    GEO Summary  \
0                   ABX Air                          GB       Domestic   
1                   ABX Air                          GB       Domestic   
2              ATA Airlines                          TZ       Domestic   
3              ATA Airlines                          TZ       Domestic   
4                Air Canada                          AC  International   
...                     ...                         ...            ...   
57376  Vietnam Airlines JSC                          VN  International   
57377       Virgin Atlantic                          VS  International   
57378      WestJet Airlines                          WS  International   
57379      WestJet Airlines                          WS  International   
57380      WestJet Airlines                          WS  International   

      GEO Region Landing Aircraft Type Aircraft Body Type  \
0             US             Freighter        Narrow Body   
1             US             Freighter        Narrow Body   
2             US             Passenger        Narrow Body   
3             US             Passenger        Narrow Body   
4         Canada             Passenger          Wide Body   
...          ...                   ...                ...   
57376       Asia             Passenger          Wide Body   
57377     Europe             Passenger          Wide Body   
57378     Canada             Passenger        Narrow Body   
57379     Canada             Passenger        Narrow Body   
57380     Canada             Passenger        Narrow Body   

      Aircraft Manufacturer Aircraft Model Aircraft Version  Landing Count  \
0         McDonnell Douglas           DC-9               30             40   
1         McDonnell Douglas           DC-9               41              1   
2                    Boeing            757              200              2   
3                    Boeing            757              300            167   
4                    Boeing            767              333              1   
...                     ...            ...              ...            ...   
57376                Airbus           A359                -             17   
57377                Boeing           B789                -             31   
57378                Boeing           B38M                -              5   
57379                Boeing           B737                -              5   
57380                Boeing           B738                -              3   

       Total Landed Weight  
0                  4066000  
1                   102000  
2                   396000  
3                 37408000  
4                   320000  
...                    ...  
57376              7758069  
57377             13175000  
57378               764000  
57379               646000  
57380               438900  

[57381 rows x 14 columns]

In [63]:
#Now we have the data ready fetched from mongoDB.
#Lets find more about the data
#Information
Data_from_db.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 57381 entries, 0 to 57380
Data columns (total 14 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   Activity Period              57381 non-null  int64 
 1   Operating Airline            57381 non-null  object
 2   Operating Airline IATA Code  57125 non-null  object
 3   Published Airline            57381 non-null  object
 4   Published Airline IATA Code  57125 non-null  object
 5   GEO Summary                  57381 non-null  object
 6   GEO Region                   57381 non-null  object
 7   Landing Aircraft Type        57381 non-null  object
 8   Aircraft Body Type           57381 non-null  object
 9   Aircraft Manufacturer        57364 non-null  object
 10  Aircraft Model               57381 non-null  object
 11  Aircraft Version             55065 non-null  object
 12  Landing Count                57381 non-null  int64 
 13  Total Landed Weight          57

In [64]:
#Getting the summary of the dataset
Data_from_db.describe(include=[object, np.number])

Activity Period Operating Airline Operating Airline IATA Code  \
count      57381.000000             57381                       57125   
unique              NaN               121                         106   
top                 NaN   United Airlines                          UA   
freq                NaN              8341                       12009   
mean      201384.769540               NaN                         NaN   
std          494.506526               NaN                         NaN   
min       200507.000000               NaN                         NaN   
25%       201001.000000               NaN                         NaN   
50%       201406.000000               NaN                         NaN   
75%       201806.000000               NaN                         NaN   
max       202212.000000               NaN                         NaN   

       Published Airline Published Airline IATA Code    GEO Summary  \
count              57381                       57125          57381   
unique               111                          97              2   
top      United Airlines                          UA  International   
freq                9608                       14156          30202   
mean                 NaN                         NaN            NaN   
std                  NaN                         NaN            NaN   
min                  NaN                         NaN            NaN   
25%                  NaN                         NaN            NaN   
50%                  NaN                         NaN            NaN   
75%                  NaN                         NaN            NaN   
max                  NaN                         NaN            NaN   

       GEO Region Landing Aircraft Type Aircraft Body Type  \
count       57381                 57381              57381   
unique         10                     3                  4   
top            US             Passenger          Wide Body   
freq        27296                 50787              27511   
mean          NaN                   NaN                NaN   
std           NaN                   NaN                NaN   
min           NaN                   NaN                NaN   
25%           NaN                   NaN                NaN   
50%           NaN                   NaN                NaN   
75%           NaN                   NaN                NaN   
max           NaN                   NaN                NaN   

       Aircraft Manufacturer Aircraft Model Aircraft Version  Landing Count  \
count                  57364          57381            55065   57381.000000   
unique                    18            104               90            NaN   
top                   Boeing            737                -            NaN   
freq                   34485           6269            27160            NaN   
mean                     NaN            NaN              NaN     108.216082   
std                      NaN            NaN              NaN     239.264079   
min                      NaN            NaN              NaN       1.000000   
25%                      NaN            NaN              NaN      13.000000   
50%                      NaN            NaN              NaN      30.000000   
75%                      NaN            NaN              NaN      78.000000   
max                      NaN            NaN              NaN    2245.000000   

        Total Landed Weight  
count          5.738100e+04  
unique                  NaN  
top                     NaN  
freq                    NaN  
mean           1.841790e+07  
std            2.941384e+07  
min            6.850000e+03  
25%            2.948000e+06  
50%            9.313000e+06  
75%            1.933913e+07  
max            2.758400e+08

In [65]:
#lets check if there are any duplicate values present in our dataset
duplicate_rows = Data_from_db.duplicated()
print(Data_from_db[duplicate_rows])

       Activity Period     Operating Airline Operating Airline IATA Code  \
132             200507               ABX Air                          GB   
223             200507               ABX Air                          GB   
531             200507          ATA Airlines                          TZ   
1288            200507          ATA Airlines                          TZ   
1685            200507            Air Canada                          AC   
...                ...                   ...                         ...   
56435           202206  Vietnam Airlines JSC                          VN   
56436           202206       Virgin Atlantic                          VS   
56437           202206      WestJet Airlines                          WS   
56438           202206      WestJet Airlines                          WS   
56439           202206      WestJet Airlines                          WS   

          Published Airline Published Airline IATA Code    GEO Summary  \
132          

Looking at the above result, we could see there are around 28000 duplicate rows, we should remove all those rows because those will lead us to the wrong assumptions and results.

In [66]:
#Deleting the duplicate rows
Data_from_db.drop_duplicates(inplace=True)
#shape of the dataset after deleting the duplicates
print(Data_from_db.shape)

(29243, 14)


In [67]:
#lets find out the emty and missing values
print('Missing Values by Count: \n\n',
      Data_from_db.isnull().sum().sort_values(ascending=True),'\n\nMissing Values by %:\n\n',
      Data_from_db.isnull().sum().sort_values(ascending=True)/Data_from_db.shape[0] * 100)

Missing Values by Count: 

 Activity Period                   0
Operating Airline                 0
Published Airline                 0
GEO Summary                       0
GEO Region                        0
Landing Aircraft Type             0
Aircraft Body Type                0
Aircraft Model                    0
Landing Count                     0
Total Landed Weight               0
Aircraft Manufacturer            17
Operating Airline IATA Code     172
Published Airline IATA Code     172
Aircraft Version               1158
dtype: int64 

Missing Values by %:

 Activity Period                0.000000
Operating Airline              0.000000
Published Airline              0.000000
GEO Summary                    0.000000
GEO Region                     0.000000
Landing Aircraft Type          0.000000
Aircraft Body Type             0.000000
Aircraft Model                 0.000000
Landing Count                  0.000000
Total Landed Weight            0.000000
Aircraft Manufacturer         

So we have missing values in following:
Aircraft Manufacturer            17,
Operating Airline IATA Code     172,
Published Airline IATA Code     172,
Aircraft Version               1158

in this we can ignore "Operating Airline IATA Code" and "Published Airline IATA Code" as that column is not much useful to us from the analysis point of view, so we can delete those columns itself.
In "Aircraft Manufacturer" there only 17 rows missing and its almost 0.05%, so we can just remove those rows.
In "Aircraft Version", there are so many missing values in this and as we need to fill it with some values.
Since its a categorical value, we can replace it with Mode value.

In [68]:
#First will remove "Operating Airline IATA Code" and "Published Airline IATA Code" columns
Data_from_db = Data_from_db.drop(columns=["Operating Airline IATA Code", "Published Airline IATA Code"])

In [69]:
#Removing the rows where "Aircraft Manufacturer" is null
Data_from_db = Data_from_db.dropna(subset=['Aircraft Manufacturer'])

In [70]:
#Since "Aircraft Version" a categorical value, we can replace it with Mode value.
#Finding the mode
mode_value = Data_from_db['Aircraft Version'].mode()[0]
#Replacing the null values with mode
Data_from_db['Aircraft Version'] = Data_from_db['Aircraft Version'].fillna(mode_value)

In [71]:
#Now lets check if there are missing values, just for confirmation.
print('Missing Values by Count: \n\n',
      Data_from_db.isnull().sum().sort_values(ascending=True),'\n\nMissing Values by %:\n\n',
      Data_from_db.isnull().sum().sort_values(ascending=True)/Data_from_db.shape[0] * 100)

Missing Values by Count: 

 Activity Period          0
Operating Airline        0
Published Airline        0
GEO Summary              0
GEO Region               0
Landing Aircraft Type    0
Aircraft Body Type       0
Aircraft Manufacturer    0
Aircraft Model           0
Aircraft Version         0
Landing Count            0
Total Landed Weight      0
dtype: int64 

Missing Values by %:

 Activity Period          0.0
Operating Airline        0.0
Published Airline        0.0
GEO Summary              0.0
GEO Region               0.0
Landing Aircraft Type    0.0
Aircraft Body Type       0.0
Aircraft Manufacturer    0.0
Aircraft Model           0.0
Aircraft Version         0.0
Landing Count            0.0
Total Landed Weight      0.0
dtype: float64


We have dealt with missing values and now our data is ready for next steps and process.

In [72]:
#If we observe the data carefully,"Operating_Airline', 'Published_Airline" are basically same, so we can remove one of the column
Data_from_db = Data_from_db.drop(columns=["Published Airline"])

In [73]:
#removing the spaces from column names to make sure that they dont create problem whilec inserting into database.
Data_from_db.rename(columns=lambda x: x.replace(' ', '_'), inplace=True)
print(Data_from_db.columns)

Index(['Activity_Period', 'Operating_Airline', 'GEO_Summary', 'GEO_Region',
       'Landing_Aircraft_Type', 'Aircraft_Body_Type', 'Aircraft_Manufacturer',
       'Aircraft_Model', 'Aircraft_Version', 'Landing_Count',
       'Total_Landed_Weight'],
      dtype='object')


In [74]:
Data_from_db.shape
Data_from_db.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 29226 entries, 0 to 57380
Data columns (total 11 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Activity_Period        29226 non-null  int64 
 1   Operating_Airline      29226 non-null  object
 2   GEO_Summary            29226 non-null  object
 3   GEO_Region             29226 non-null  object
 4   Landing_Aircraft_Type  29226 non-null  object
 5   Aircraft_Body_Type     29226 non-null  object
 6   Aircraft_Manufacturer  29226 non-null  object
 7   Aircraft_Model         29226 non-null  object
 8   Aircraft_Version       29226 non-null  object
 9   Landing_Count          29226 non-null  int64 
 10  Total_Landed_Weight    29226 non-null  int64 
dtypes: int64(3), object(8)
memory usage: 2.7+ MB


In [75]:
#Now lets to connect to POSTGRESQL and store the data in the table.
import psycopg2

#connection parameters
Host_name = 'localhost'
Port_number = '5432'  
Database_name = 'Flight_details'  
DB_username = 'postgres'  
DB_password = '8722708378'  

#Connect to the PostgreSQL server
connection = psycopg2.connect(host=Host_name, port=Port_number, dbname=Database_name, user=DB_username, password=DB_password)

# Open a cursor to execute SQL queries
curs = connection.cursor()

#query for creating a table
curs.execute("CREATE TABLE Air_Traffic_Landing_Statistics(Activity_Period         varchar(100),Operating_Airline       varchar(100),GEO_Summary             varchar(100),GEO_Region              varchar(100),Landing_Aircraft_Type   varchar(100),Aircraft_Body_Type      varchar(100),Aircraft_Manufacturer   varchar(100),Aircraft_Model          varchar(100),Aircraft_Version        varchar(100),Landing_Count           varchar(100),Total_Landed_Weight     varchar(100)   ); ")

#inserting the data into postgresql table
for i, row in Data_from_db.iterrows():
    values = tuple(row.values)
    insert_sql = "INSERT INTO Air_Traffic_Landing_Statistics (Activity_Period,Operating_Airline,GEO_Summary,GEO_Region,Landing_Aircraft_Type,Aircraft_Body_Type,Aircraft_Manufacturer,Aircraft_Model,Aircraft_Version,Landing_Count,Total_Landed_Weight) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)"
    curs.execute(insert_sql, values)
    
#Committing the transaction
connection.commit()

# Closing the cursor and the connection
curs.close()
connection.close()

DuplicateTable: relation "air_traffic_landing_statistics" already exists


In [ ]:
#Now lets fetch the data from the database

#connection parameters
Host_name = 'localhost'
Port_number = '5432'  
Database_name = 'Flight_details'  
DB_username = 'postgres'  
DB_password = '8722708378'  

#Connect to the PostgreSQL server
connection = psycopg2.connect(host=Host_name, port=Port_number, dbname=Database_name, user=DB_username, password=DB_password)

#query for fetching all the data from the table
query="select * from Air_Traffic_Landing_Statistics;"

#fetching the data using a query
fetched_data = pd.read_sql(query, connection)    

#Closing the connection
connection.close()

In [ ]:
fetched_data.shape

In [ ]:
fetched_data.head()

#Transformation
If we look at the data, activity_period is actually a year with month,so we can change the format of that column to datetime.
Also we can have 2 separate columns each for a year and a month which may be useful for us in the future for visualuzation like monthly or yearly flights and so on.

In [ ]:
#First will create 2 separate columns for month and year

#creating a new column 'Year' by extracting the first four characters from 'activity_period'
fetched_data['Year'] = fetched_data['activity_period'].str[:4]

#creating a new column 'Month' by extracting the last two characters from 'activity_period'
fetched_data['Month'] = fetched_data['activity_period'].str[-2:]

fetched_data.head()

In [ ]:
#Will change the format of the activity_period.Landing_count,Total_landed_weight
fetched_data['activity_period'] = pd.to_datetime(fetched_data['activity_period'], format='%Y%m')
#Will change the format of Landing_count,Total_landed_weight as well as those are the numbers
fetched_data['landing_count']=fetched_data['landing_count'].astype(int)
fetched_data['total_landed_weight']=fetched_data['total_landed_weight'].astype(int)

fetched_data.head()

Now our data is ready and we can proceed with Visualization

In [ ]:
#Visualization
#Bar chart showing total number of landings by operating airline
import matplotlib.pyplot as plt

## group data by operating airline and sum the number of landings
airline_landing_counts = fetched_data.groupby('operating_airline')['landing_count'].sum()

# sort the values in descending order and take the top 10
top_10_airlines = airline_landing_counts.sort_values(ascending=False)[:10]

# create a bar chart showing the number of landings by airline
plt.figure(figsize=(12,6))
plt.bar(top_10_airlines.index, top_10_airlines.values)
plt.xticks(rotation=45, ha='right')
plt.xlabel('Operating Airline')
plt.ylabel('Number of Landings')
plt.title('Top 10 Operating Airlines by with Highest Number of Landings')
plt.show()

In [ ]:
#Pie chart showing percentage of landings by aircraft body type
# Create a groupby object to aggregate landings by aircraft body type
landings_by_body_type = fetched_data.groupby('aircraft_body_type')['landing_count'].sum()

# Sort the values in descending order
landings_by_body_type = landings_by_body_type.sort_values(ascending=False)

# Create a pie chart
plt.figure(figsize=(6,6))
plt.pie(landings_by_body_type.values, labels=landings_by_body_type.index, autopct='%1.1f%%')
plt.title('Percentage of landings by aircraft body type')
plt.show()


In [ ]:
#Line chart showing trend of total landed weight over time
# Create a groupby object to aggregate total landed weight by year and month
landed_weight_by_year = fetched_data.groupby(['Year'])['total_landed_weight'].sum()

# Create a line chart
plt.figure(figsize=(10,6))
plt.plot(landed_weight_by_year.index, landed_weight_by_year.values, color='red')
plt.title('Trend of total landed weight over time')
plt.xlabel('Year')
plt.ylabel('Total landed weight')
plt.xticks(rotation=90)
plt.show()


In [ ]:
#Bar chart showing total number of landings by aircraft manufacturer
# Create a groupby object to aggregate landings by aircraft manufacturer
landings_by_manufacturer = fetched_data.groupby('aircraft_manufacturer')['landing_count'].sum()

# Sort the values in descending order
landings_by_manufacturer = landings_by_manufacturer.sort_values(ascending=False)

# Create a bar chart
plt.figure(figsize=(10,6))
plt.bar(landings_by_manufacturer.index, landings_by_manufacturer.values, color='green')
plt.title('Total number of landings by aircraft manufacturer')
plt.xlabel('Aircraft Manufacturer')
plt.ylabel('Total number of landings')
plt.xticks(rotation=90)
plt.show()

In [ ]:
#Heatmap of the number of landings by year and month
import seaborn as sns
# Create a pivot table to aggregate landings by year and month
landings_by_date = pd.pivot_table(fetched_data, values='landing_count', index='Year', columns='Month', aggfunc='sum')

# Create a heatmap
plt.figure(figsize=(10,6))
sns.heatmap(landings_by_date, cmap='YlGnBu')
plt.title('Number of landings by year and month')
plt.xlabel('Month')
plt.ylabel('Year')
plt.show()


In [ ]:
#Pie chart showing the distribution of landings by geo summary
landings_by_geo_summary = fetched_data.groupby(['geo_summary'])['landing_count'].sum()

plt.figure(figsize=(6,6))
plt.pie(landings_by_geo_summary.values, labels=landings_by_geo_summary.index, autopct='%1.1f%%')
plt.title('Distribution of landings by geo summary')
plt.show()

In [ ]:
#Bar chart showing the total number of landings by geo region
#Creating a groupby object to aggregate total landings by geo region
landings_by_region = fetched_data.groupby(['geo_region'])['landing_count'].sum()

#Sorting the values in descending order
landings_by_region = landings_by_region.sort_values(ascending=False)

# Create a bar chart
plt.figure(figsize=(8,6))
plt.bar(landings_by_region.index, landings_by_region.values,color='gold')
plt.title('Total number of landings by Geo Region')
plt.xlabel('Geo region')
plt.ylabel('Number of landings')
plt.xticks(rotation=45)
plt.show()

In [ ]:
#line chart showing the trend of total landed weight over time by landing aircraft type
#Creating a groupby object to aggregate total landed weight by landing aircraft type and year
landed_weight_by_aircraft_and_year = fetched_data.groupby(['landing_aircraft_type', 'Year'])['total_landed_weight'].sum()

#Creating a line chart for each aircraft type
aircraft_types = fetched_data['landing_aircraft_type'].unique()
colors = ['blue', 'green', 'red', 'purple', 'orange', 'brown', 'pink', 'gray', 'olive', 'cyan']
plt.figure(figsize=(10, 6))
for i, aircraft_type in enumerate(aircraft_types):
    subset = landed_weight_by_aircraft_and_year.loc[aircraft_type]
    plt.plot(subset.index, subset.values, label=aircraft_type, color=colors[i])
plt.title('Trend of total landed weight over time by landing aircraft type')
plt.xlabel('Year')
plt.ylabel('Total landed weight')
plt.legend()
plt.show()

In [ ]:
fetched_data.to_csv('mydata.csv', index=False)